sdlf-ddk-lightweight/data_lake/src/glue/pyshell_scripts/sdlf_heavy_transform/demoteam/legislators/main.py

```diff
@@ -58,13 +63,28 @@
                      Join.apply(persons, memberships, 'id', 'person_id'),
                      'org_id', 'organization_id').drop_fields(['person_id', 'org_id'])
 
-persons.toDF().write.mode("overwrite").parquet(
-    '{}/persons/'.format(destination))
-organizations.toDF().write.mode("overwrite").parquet(
-    '{}/organizations/'.format(destination))
-memberships.toDF().write.mode("overwrite").parquet(
-    '{}//memberships/'.format(destination))
-history.toDF().write.mode("overwrite").parquet(
-    '{}/history/'.format(destination), partitionBy=['org_name'])
+def overwrite_table(frame: DynamicFrame, db: str, table: str, dest_path: str, partitions: List[str] = []):
```

Comment:
Just curious what was the reasoning here to replace crawlers with the job updating the catalog itself?

Reply:
Keeping 1 crawler per dataset felt like unnecessary additional infra and could lead to issues for high volume use cases where crawler run time is longer than 5 minutes

Seems more efficient to handle the cataloging within the glue job itself

Reply:
I see. I've seen implementations where we also removed the crawlers leveraging AWS SDK for pandas for data transformation and cataloguing (albeit it was only light transforms in that case) and I agree the less infrastructure to manage the better. Thanks.

Reply:
Another reason to remove the crawler is a bit more foundational. The crawler was defined as part of the pipeline, but it's functionality is actually specific to one dataset (one pipeline can have multiple datasets). There are some not-so-pretty workarounds to this if we add some functionality to the `register_dataset()` method, but the easiest (and more performant) method is to just remove the crawler altogether and replace it with updating the catalog within each glue job.